### ISSUE TITLE
use spark-filetransfer without distributed filesystem

### ISSUE THREAD

**Issue by morind** [arcizon/spark-filetransfer#10]
Hello,

I try to use this library in Spark cluster without distributed filesystem (only local disk) but I can't.
I try to use S3 storage but I have this error :

```
21/03/24 11:12:07 INFO SFTP: Downloading file from /data/test.csv to /opt/spark/workdir/spark-filetransfer-ce178b55-4a23-4580-b281-8a45293a59a2/test.csv
21/03/24 11:12:07 INFO DfsUtils: Copying file from local path /opt/spark/workdir/spark-filetransfer-ce178b55-4a23-4580-b281-8a45293a59a2 to dfs path s3a://spark-data/d1e9b6fa-d6c0-4459-ae36-e46060f1476
..
py4j.protocol.Py4JJavaError: An error occurred while calling o134.load.
: java.io.FileNotFoundException: Not a file: /opt/spark/workdir/spark-filetransfer-ce178b55-4a23-4580-b281-8a45293a59a2

Here the code:

```
    df = spark.read \
        .format("filetransfer") \
        .option("protocol", "sftp") \
        .option("host", "XXXX") \
        .option("port", "21") \
        .option("username", "XXXX") \
        .option("password", "XXXX") \
        .option("fileFormat", "csv") \
        .option("delimiter", ";") \
        .option("inferSchema", "true") \
        .option("header", "true") \
        .option("localTempPath", "/opt/spark/workdir/") \
        .option("dfsTempPath", "s3a://spark-data/") \
        .load("/data/test.csv") \
        .show()
``
I've tried lot of things but still I'm stuck.

**Comment by shaikmanu797** [comment]
@morind, I haven't actually tested with s3 earlier I will try to replicate your setup and fix issues for a successful run.

In the meantime, could you let me know if the /opt/spark/workdir/spark-filetransfer-ce178b55-4a23-4580-b281-8a45293a59a2/test.csv exists after downloading the file from SFTP server?

**Comment by morind** [comment]
@shaikmanu797 Thanks for your reply.
Ok I replay my test and check the existence.
Otherwise, I've also tried with .config("spark.hadoop.fs.defaultFS", "s3a://spark-data/") .. in order to have a global Fs based on S3. Just for test. But still I have the same error.
But, at the end, if you have a setup that works with a Spark cluster with only local storage I would appreciate to have it. Even with only one Driver/Executor it would be enough.
Thanks a lot for your help

**Comment by shaikmanu797** [comment]
@morind Could you confirm me if the process is successful in downloading the file to local filesystem (localTempPath) ?

I understand that following part failing where the process tries to upload the download file into DFSTempPath.

**Comment by morind** [comment]
@shaikmanu797 
Yes I confirm the file is well downloaded (content ok) in /opt/spark/workdir/spark-filetransferXXX (localTempPath) on the Driver side

**Comment by morind** [comment]
@shaikmanu797 Just for information, if I don't use S3 as distributed Fs and I use only local storage I have this error: "java.io.FileNotFoundException: File file:/tmp/spark/workdir/a34eaad3-82fc-49be-b9f9-6373c93ea159/test.csv does not exist"
And just before: 
"21/03/24 15:07:24 INFO SFTP: Downloading file from /data/test.csv to /tmp/spark/workdir/spark-filetransfer-ceadaf4f-4cdb-4905-a84c-46fe67a469bb/test.csv
21/03/24 15:07:25 INFO DfsUtils: Copying file from local path /tmp/spark/workdir/spark-filetransfer-ceadaf4f-4cdb-4905-a84c-46fe67a469bb to dfs path /tmp/spark/workdir/a34eaad3-82fc-49be-b9f9-6373c93ea159"
